### ISSUE TITLE
Imports used exclusively in the `<template>` part of Vue SFCs are considered "never used" by Volar

### ISSUE THREAD

**Issue by simensol** [johnsoncodehk/volar#1757]
This issue was [originally raised](https://github.com/simonhaenisch/prettier-plugin-organize-imports/issues/71) in [simonhaenisch/prettier-plugin-organize-imports](https://github.com/simonhaenisch/prettier-plugin-organize-imports), a library which sorts imports and removes unused ones.

_Originally posted by @simonhaenisch in https://github.com/simonhaenisch/prettier-plugin-organize-imports/issues/71#issuecomment-1224104071_:

> [`simonhaenisch/prettier-plugin-organize-imports`] is using `@volar/vue-typescript` as a drop-in replacement to create the language service, and after that it's just a matter of calling the `organizeImports` method.

It seems that imports exclusively used in the `<template>` part of Vue SFCs are incorrectly considered as "never used" by Volar. Here is a minimal working example:

`package.json`:
```json
...
    "devDependencies": {
        "@volar/vue-typescript": "^0.40.1",
        "prettier": "^2.7.1",
        "prettier-plugin-organize-imports": "^3.1.0",
...
```

`types.ts`:
```typescript
export enum ThemeColor {
    PRIMARY = "primary",
    SECONDARY = "secondary",
    ACCENT = "accent",
    ERROR = "error",
    INFO = "info",
    SUCCESS = "success",
    WARNING = "warning"
}
```

`sfc.vue`:
```vue
<script setup lang="ts">
import { ThemeColor } from "@/types"
</script>

<template>
    Color is: {{ ThemeColor.PRIMARY }}
</template>
```

`import { ThemeColor } from "@/types"` is incorrectly considered as "never used" by Volar and then removed by `simonhaenisch/prettier-plugin-organize-imports`. If I add `const color = ThemeColor.PRIMARY` to the `<script setup>` part of `sfc.vue`, the import is not considered "never used" by Volar.

The same happens with functions used exclusively in the `<template>` part of Vue SFCs:

`helpers.ts`:
```typescript
export function helloWorld(): string {
    return "Hello Worlds"
}
```

`sfc.vue`:
```vue
<script setup lang="ts">
import { helloWorld } from "@/helpers"
</script>

<template>
    {{ helloWorld() }}
</template>
```

Here, `import { helloWorld } from "@/helpers"` is marked as "never used" by Volar. However, if I add `const hello = helloWorld()` to the `<script setup>` part, the import is kept untouched.

**Comment by simonhaenisch** [comment]
Could be related to path aliases as well btw? Have you tried whether it's reproducible with a relative path instead of `@/`?

**Comment by simensol** [comment]
Same isse with

```vue
<script setup lang="ts">
import { ThemeColor } from "../utilities/types"
</script>

<template>Color is: {{ ThemeColor.PRIMARY }}</template>
```

**Comment by craigrileyuk** [comment]
I'm getting the same problem too, and I'm not directly using Typescript in my project.

**Comment by johnsoncodehk** [comment]
Duplicate of #1729

**Comment by simonhaenisch** [comment]
Thanks for the quick fix! Will this be released as `0.40.2` or `0.41.0`?

**Comment by johnsoncodehk** [comment]
@simonhaenisch it's v0.40.2 because there are no breaking changes.
